python/imageutils.py

```diff
@@ -233,73 +232,103 @@ def clustered_read(self, reconstructed, legend, mem):
 
     def multiple_reads(self, reconstructed, legend, mem):
         """ Reconstruct an image given a set of splits and amount of available memory.
-        Multiple_reads: load splits servel times to read a complete slice
+        multiple_reads: load splits servel times to read a complete slice
         Currently it can work on random shape of splits and in unsorted order
         :param reconstructed: the fileobject pointing to the to-be reconstructed image
-        :param legend: containing the URIs of the splits. 
-        :param mem: amount of available memory in bytes. 
+        :param legend: containing the URIs of the splits.
+        :param mem: bytes to be written into the file
         """
-        # The basic idea is that calculate how much bytes it needs to write to a file based
-        # on mem parameter. Then filter all the position (offset) of tiles in that range and add them to data_dict
-        # After all the blocks have been read, write data_dict to file. And start next round...
-        # until the file has done writing.
-        # get original image infomation
-        rec_dims = self.header.get_data_shape()
-        y_size = rec_dims[0]
-        z_size = rec_dims[1]
-        x_size = rec_dims[2]
+        Y_size, Z_size, X_size = self.header.get_data_shape()
         bytes_per_voxel = self.header['bitpix'] / 8
-        img_size = self.header.single_vox_offset + bytes_per_voxel * (x_size * y_size * z_size)
-        print y_size, z_size, x_size
+        header_offset = self.header.single_vox_offset
+        reconstructed_img_voxels = X_size * Y_size * Z_size
 
-        total_read = 0
-        total_seek = 0
-        total_defrag = 0
-        total_write = 0
+        # for benchmarking
+        total_read_time = 0
+        total_seek_time = 0
+        total_write_time = 0
+        total_seek_number = 0
```

Comment:
It would be very useful to also measure the number of seeks (not just the time), to compare it with the formulas in the paper.